File: KOPracticeWithApi/wwwroot/js/VM/ItemModel.js

function ItemViewModel(data) {
    var self = this;
    self.id = ko.observable(data.id || 0);
    self.name = ko.observable(data.name || '');
    self.price = ko.observable(data.price || 0);
    self.file = ko.observable(null);
    self.isPublished = ko.observable(data.isPublished || false);
    self.imageUrl = ko.observable(data.imageUrl || '');
    self.showImage = ko.observable(false);
}


ItemViewModel.fromServerModel = function (data) {
    return new ItemViewModel(data)
};

File: KOPracticeWithApi.Models/ItemModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace KOPracticeWithApi.Models
{
    public class ItemModel
    {
        [Key]
        public int Id { get; set; }
        [StringLength(50)]
        public string Name { get; set; }
        [Column(TypeName = "Decimal(18,2)")]
        
        public decimal Price { get; set; }
        public string? ImageUrl { get; set; }
        public bool IsPublished { get; set; }
    }
}
